src/kOS.Safe/Communication/Connection.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using kOS.Safe.Encapsulation;
+using kOS.Safe.Encapsulation.Suffixes;
+
+namespace kOS.Safe.Communication
+{
+    [kOS.Safe.Utilities.KOSNomenclature("Connection")]
+    public abstract class Connection<S> : Structure where S : SharedObjects
```

Comment:
This template usage ``<S>``seems like a very strange thing to do when I don't think ``SharedObjects`` has any other classes derived from it, or planned to be derived from it.  Or to put it another way, what are you expecting someone might use in place of the ``S`` here other than ``SharedObjects`` ?